File: fix_colors.js
const fs = require('fs');
const path = require('path');

function fixFile(filepath) {
    const origContent = fs.readFileSync(filepath, 'utf8');
    let content = origContent;

    // Replace color: Colors.white with color: Theme.of(context).colorScheme.onSurface
    content = content.replace(/color:\s*Colors\.white\s*(?=[,\)])/g, 'color: Theme.of(context).colorScheme.onSurface');

    if (content !== origContent) {
        let lines = content.split('\n');
        for (let i = 0; i < lines.length; i++) {
            if (lines[i].includes('Theme.of(context)')) {
                // remove const before widgets
                lines[i] = lines[i].replace(/const\s+(TextStyle|Icon|Text|Column|Row|Expanded|Padding|SliverToBoxAdapter|Center|Align)/g, '$1');
            }
        }
        content = lines.join('\n');
        fs.writeFileSync(filepath, content, 'utf8');
        console.log(`Updated ${filepath}`);
    }
}

function walk(dir) {
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        const filepath = path.join(dir, file);
        const stat = fs.statSync(filepath);
        if (stat && stat.isDirectory()) {
            walk(filepath);
        } else {
            if (filepath.endsWith('.dart')) {
                fixFile(filepath);
            }
        }
    });
}

walk('c:\\\\Mentron_ap\\\\mentron_new\\\\mentron_flutter\\\\lib\\\\features');

File: fix_colors.py
import os
import re

def fix_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    orig_content = content

    # Replace `color: Colors.white` -> `color: Theme.of(context).colorScheme.onSurface`
    # only if NOT followed by `.withOpacity` or `.withValues`
    # Also handle `Colors.white,` and `Colors.white)`
    
    content = re.sub(r'color:\s*Colors\.white\s*(?=[,\)])', r'color: Theme.of(context).colorScheme.onSurface', content)

    # Now we have Theme.of(context) in places that might be marked const.
    lines = content.split('\n')
    for i, line in enumerate(lines):
        if 'Theme.of(context)' in line:
            line = re.sub(r'const\s+(TextStyle|Icon|Text|Column|Row|Expanded|Padding|SliverToBoxAdapter|Center|Align)', r'\1', line)
            lines[i] = line
            
    content = '\n'.join(lines)

    if content != orig_content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated {filepath}")

for root, dirs, files in os.walk(r'c:\Mentron_ap\mentron_new\mentron_flutter\lib\features'):
    for file in files:
        if file.endswith('.dart'):
            fix_file(os.path.join(root, file))
